kani-compiler/src/kani_middle/points_to/points_to_graph.rs

```diff
@@ -0,0 +1,216 @@
+        }
+    }
+
+    /// Collect all scalar places to which a given place can alias. This is needed to resolve all
+    /// dereference projections.
+    pub fn follow_from_place(
+        &self,
+        place: Place<'tcx>,
+        current_def_id: DefId,
+    ) -> HashSet<GlobalMemLoc<'tcx>> {
+        let place_or_alloc: LocalMemLoc =
+            Place { local: place.local, projection: List::empty() }.into();
+        let mut node_set = HashSet::from([place_or_alloc.with_def_id(current_def_id)]);
+        for projection in place.projection {
+            match projection {
+                ProjectionElem::Deref => {
+                    node_set = self.follow(&node_set);
+                }
+                ProjectionElem::Field(..)
+                | ProjectionElem::Index(..)
+                | ProjectionElem::ConstantIndex { .. }
+                | ProjectionElem::Subslice { .. }
+                | ProjectionElem::Downcast(..)
+                | ProjectionElem::OpaqueCast(..)
+                | ProjectionElem::Subtype(..) => {
+                    /* There operations are no-ops w.r.t aliasing since we are tracking it on per-object basis. */
+                }
+            }
+        }
+        node_set
+    }
+
+    /// Dump the graph into a file using the graphviz format for later visualization.
+    pub fn dump(&self, file_path: &str) {
+        let mut nodes: Vec<String> =
+            self.edges.keys().map(|from| format!("\t\"{:?}\"", from)).collect();
+        nodes.sort();
+        let nodes_str = nodes.join("\n");
+
+        let mut edges: Vec<String> = self
+            .edges
+            .iter()
+            .flat_map(|(from, to)| {
+                let from = format!("\"{:?}\"", from);
+                to.iter().map(move |to| {
+                    let to = format!("\"{:?}\"", to);
+                    format!("\t{} -> {}", from.clone(), to)
+                })
+            })
+            .collect();
+        edges.sort();
+        let edges_str = edges.join("\n");
+
+        std::fs::write(file_path, format!("digraph {{\n{}\n{}\n}}", nodes_str, edges_str)).unwrap();
+    }
+
+    /// Find a transitive closure of the graph starting from a set of given locations; this also
+    /// includes statics.
+    pub fn transitive_closure(&self, targets: HashSet<GlobalMemLoc<'tcx>>) -> PointsToGraph<'tcx> {
+        let mut result = PointsToGraph::empty();
+        // Working queue.
+        let mut queue = VecDeque::from_iter(targets);
+        // Add all statics, as they can be accessed at any point.
+        let statics = self.edges.keys().filter(|node| matches!(node, GlobalMemLoc::Global(_)));
+        queue.extend(statics);
+        // Add all entries.
+        while !queue.is_empty() {
+            let next_target = queue.pop_front().unwrap();
+            result.edges.entry(next_target).or_insert_with(|| {
+                let outgoing_edges =
+                    self.edges.get(&next_target).cloned().unwrap_or(HashSet::new());
```

Comment:
wait.. why are you invoking `get` in here? This will only execute if there is no entry.

Reply:
It's `result.edges` vs `self.edges`, the first one does not have the entry, but the second one could